contingency/contingency-api/src/main/java/com/powsybl/contingency/contingency/list/IdentifierContingencyList.java

```diff
@@ -52,10 +52,12 @@ public List<NetworkElementIdentifier> getIdentifiants() {
     @Override
     public List<Contingency> getContingencies(Network network) {
         return networkElementIdentifiers.stream()
-                .map(identifiant -> identifiant.filterIdentifiable(network))
-                .filter(Optional::isPresent)
-                .map(identifiable -> new Contingency(identifiable.get().getId(),
-                        ContingencyElement.of(identifiable.get())))
+                .map(identifiant -> Pair.of(identifiant.getContingencyName(), identifiant.filterIdentifiable(network)))
+                .filter(identifiant -> !identifiant.getSecond().isEmpty())
+                .map(identifiant -> new Contingency(identifiant.getFirst(), identifiant.getSecond()
```

Comment:
```suggestion
               .filter(identifier -> !identifier.filterIdentifiable(network).isEmpty())
               .map(identifier -> new Contingency(identifier.getContingencyName(), identifier.filterIdentifiable(network))

Reply:
Contingency constructor does not take a identifiable set but with what you write I think this is more accurate
```java
.filter(identifier -> !identifier.filterIdentifiable(network).isEmpty())
                .map(identifier -> new Contingency(identifier.getContingencyName(), 
                        identifier.filterIdentifiable(network)
                                .stream()
                                .map(ContingencyElement::of)
                                .collect(Collectors.toList())))
```